server/jobs/project_command_output_handler.go

```diff
@@ -228,39 +188,23 @@ func (p *AsyncProjectCommandOutputHandler) GetJobIdMapForPull(pullInfo PullInfo)
 	return nil
 }
 
-func (p *AsyncProjectCommandOutputHandler) CleanUp(pullInfo PullInfo) {
-	if value, ok := p.pullToJobMapping.Load(pullInfo); ok {
-		jobMapping := value.(map[string]bool)
-		for jobID := range jobMapping {
-			p.projectOutputBuffersLock.Lock()
-			delete(p.projectOutputBuffers, jobID)
-			p.projectOutputBuffersLock.Unlock()
-
-			p.receiverBuffersLock.Lock()
-			delete(p.receiverBuffers, jobID)
-			p.receiverBuffersLock.Unlock()
-		}
-
-		// Remove job mapping
-		p.pullToJobMapping.Delete(pullInfo)
-	}
-}
-
 // NoopProjectOutputHandler is a mock that doesn't do anything
 type NoopProjectOutputHandler struct{}
 
-func (p *NoopProjectOutputHandler) Send(ctx models.ProjectCommandContext, msg string, isOperationComplete bool) {
+func (p *NoopProjectOutputHandler) Send(ctx models.ProjectCommandContext, msg string) {
 }
 
-func (p *NoopProjectOutputHandler) Register(jobID string, receiver chan string)   {}
-func (p *NoopProjectOutputHandler) Deregister(jobID string, receiver chan string) {}
-
 func (p *NoopProjectOutputHandler) Handle() {
 }
 
+func (p *NoopProjectOutputHandler) Register(jobID string, receiver chan string) {}
+
 func (p *NoopProjectOutputHandler) CleanUp(pullInfo PullInfo) {
 }
 
-func (p *NoopProjectOutputHandler) IsKeyExists(key string) bool {
+func (p *NoopProjectOutputHandler) DoesKeyExist(key string) bool {
```

Comment:
Do we still need this if you removed it?

Reply:
Nope! Thanks for the catch :)